Add CRUD tests for PuppetClass entity

jyejare opened the pull request:
Add tests for creating, reading, updating and deleting PuppetClass entities. The
added tests are generic, so they have been added to module
`tests.foreman.api.test_multiple_paths`. Example test results:

    $ nosetests tests/foreman/api/test_multiple_paths.py -m PuppetClass
    ..........S
    ----------------------------------------------------------------------
    Ran 11 tests in 3.353s

    OK (SKIP=1)

Depends on SatelliteQE/nailgun#45.

tkolhar commented:
ACK

Ichimonji10 commented on tests/foreman/api/test_puppet_classes.py:
```diff
@@ -0,0 +1,102 @@
+        @Feature: Puppet Classes
+
+        """
+        puppet_class_id = entities.PuppetClass(**attrs).create_json()['id']
```
Do calls to `PuppetClass.create` and `PuppetClass.read` work?

jyejare replied:
Yes.... the read and create works and returns entities.puppetclass object but for verification i used create_json() which will get me the puppet_class id and read_json() which will return the dictionary of puppet class items.

Ichimonji10 commented:
The tests added by this pull request are all valid, but they are very generic. They just create, update and delete a puppet class. Could a couple lines be added to module `tests.foreman.api.test_multiple_paths` instead?

jyejare commented:
Thanks @Ichimonji10 for this easy to use and develop suggestion.
So as per your suggestion to add the tests in tests.foreman.api.test_multiple_paths, I have updated the appropriate tests for puppet class entity. Please review and merge.
Following are the passing logs for all the tests for PuppetClass entity:
Note: I have added the entity in alphabetical order only. Please ignore the number 1 in test logs that is just for testing purpose. 
```
============================================================================
[root@jyejare robottelo]# nosetests ./tests/foreman/api/test_multiple_paths.py -m test_get_status_code_1_Puppet
.
----------------------------------------------------------------------
Ran 1 test in 3.113s

OK

============================================================================
[root@jyejare robottelo]# nosetests -m test_get_unauthorized_1_Puppet -s ./tests/foreman/api/test_multiple_paths.py
.
----------------------------------------------------------------------
Ran 1 test in 1.692s

OK
============================================================================
[root@jyejare robottelo]# nosetests -m test_post_status_code_1_Puppet -s ./tests/foreman/api/test_multiple_paths.py
.
----------------------------------------------------------------------
Ran 1 test in 2.043s

OK
============================================================================
[root@jyejare robottelo]# nosetests -m test_post_unauthorized_1_Puppet -s ./tests/foreman/api/test_multiple_paths.py

S
----------------------------------------------------------------------
Ran 1 test in 2.041s

OK (SKIP=1)
============================================================================
[root@jyejare robottelo]# nosetests -m test_get_status_code_1_Puppet -s ./tests/foreman/api/test_multiple_paths.py
.
----------------------------------------------------------------------
Ran 1 test in 4.774s

OK
============================================================================
[root@jyejare robottelo]# nosetests -m test_put_status_code_1_Puppet -s ./tests/foreman/api/test_multiple_paths.py
.
----------------------------------------------------------------------
Ran 1 test in 3.871s

OK
============================================================================
[root@jyejare robottelo]# nosetests -m test_delete_status_code_1_Puppet -s ./tests/foreman/api/test_multiple_paths.py
.
----------------------------------------------------------------------
Ran 1 test in 4.105s

OK
============================================================================
[root@jyejare robottelo]# nosetests -m test_put_and_get_1_Puppet -s ./tests/foreman/api/test_multiple_paths.py
.
----------------------------------------------------------------------
Ran 1 test in 9.255s

OK
============================================================================
[root@jyejare robottelo]# nosetests -m test_post_and_get_1_Puppet -s ./tests/foreman/api/test_multiple_paths.py
.
----------------------------------------------------------------------
Ran 1 test in 5.783s

OK
============================================================================
[root@jyejare robottelo]# nosetests -m test_delete_and_get_1_Puppet -s ./tests/foreman/api/test_multiple_paths.py
.
----------------------------------------------------------------------
Ran 1 test in 5.895s

OK
============================================================================
[root@jyejare robottelo]# nosetests -m test_entity_read_1_Puppet -s ./tests/foreman/api/test_multiple_paths.py
.
----------------------------------------------------------------------
Ran 1 test in 3.753s

OK
============================================================================
```

Ichimonji10 commented:
ACK

Ichimonji10 commented:
FYI, you can run all puppet-related tests in that file with this command:

    nosetests tests/foreman/api/test_multiple_paths.py -m PuppetClass

jyejare commented:
Great ! Thanks @Ichimonji10

Ichimonji10 commented:
Hold on. Why are there two commits in this PR?

elyezer commented:
ACK, pending squash.

Ichimonji10 commented:
Looks great to me. @elyezer, could you take one last look at the changes in this PR to make sure that things are sane? I'm too involved in this PR to be a completely detached judge of things.

elyezer commented:
ACK
